storm-core/src/clj/backtype/storm/daemon/nimbus.clj

```diff
@@ -1103,21 +1230,39 @@
      (.set_reset_log_level_timeout_epoch log-config (coerce/to-long timeout))
      (.unset_reset_log_level_timeout_epoch log-config))))
 
+(defmethod blob-sync-code :distributed [conf nimbus]
+  (if (not (is-leader nimbus :throw-exception false))
+    (let [storm-cluster-state (:storm-cluster-state nimbus)
+          nimbus-host-port-info (:nimbus-host-port-info nimbus)
+          blob-store-key-list (into [] (get-key-set-from-blob-store (:blob-store nimbus)))
```

Comment:
It would be better to have a `nimbus/get-key-list-from-blob-store` function, and have `nimbus/get-key-set-from-blob-store` make a list out of it, rather than doing a list -> set -> list conversion.